### PR TITLE
Fixed filter when fetching torrents in your own channel

### PR DESCRIPTION
**Pull request by devos50**
Also removed some terms from the family filter list.

### CODE REVIEW

**Comment by devos50** [comment]
Dirty reactor:

```
ERROR   1477472583  test_as_server:TestBoostingManagerUtilities:129  >     <DelayedCall 0x7f608014a908 [0.875077962875s] called=0 cancelled=0 EPollReactor.callInThread(<function <lambda> at 0x7f608001cb90>)>
```

I will figure this one out as soon as I can.

**Comment by devos50** [comment]
retest this please

File: Tribler/Core/Modules/restapi/channels/channels_torrents_endpoint.py [hunk 1]
```diff
@@ -70,11 +71,15 @@ def render_GET(self, request):
...
+        if 'disable_filter' in request.args and len(request.args['disable_filter']) > 0 \
+                and request.args['disable_filter'][0] == "1":
+            should_filter = False
+
         results_json = []
         for torrent_result in results_local_torrents_channel:
             torrent_json = convert_db_torrent_to_json(torrent_result)
-            if (self.session.tribler_config.get_family_filter_enabled() and
-                    self.session.lm.category.xxx_filter.isXXX(torrent_json['name'])) or torrent_json['name'] is None:
+            if (should_filter and torrent_json['category'] == 'xxx') or torrent_json['name'] is None:
```
Comment (brussee) [inline Tribler/Core/Modules/restapi/channels/channels_torrents_endpoint.py]: should_filter must not influence torrents with empty name. However I think that these are renamed by the api to "Unnamed torrent" at the moment.

Comment (devos50) [inline Tribler/Core/Modules/restapi/channels/channels_torrents_endpoint.py]: @brussee did you notice the parenthesis? `should_filter` has no influences on the torrent name, correct me if I'm wrong.

Comment (brussee) [inline Tribler/Core/Modules/restapi/channels/channels_torrents_endpoint.py]: Yes; after this code the renaming happens if applicable.

**Pull request merged**